documentation/source/user_section/tutorials/gm-wm/improving-registration-with-gm-seg/gm-informed-t2s-template-registration.rst

```diff
@@ -0,0 +1,54 @@
+
+This registration step is similar to that of previous tutorials. However, a notable difference is that rather than using the full spinal cord segmentations for the arguments ``-iseg`` and ``-dseg``, we use the white matter segmentation. This is because its interior perimeter outlines the gray matter shape, while its exterior perimeter outlines the shape of the spinal cord itself, which better informs the registration algorithm.
+
+.. code:: sh
+
+   sct_register_multimodal -i "${SCT_DIR}/data/PAM50/template/PAM50_t2s.nii.gz" \
+                           -iseg "${SCT_DIR}/data/PAM50/template/PAM50_wm.nii.gz" \
+                           -d t2s.nii.gz \
+                           -dseg t2s_wmseg.nii.gz \
+                           -initwarp ../t2/warp_template2anat.nii.gz \
+                           -initwarpinv ../t2/warp_anat2template.nii.gz \
+                           -param step=1,type=seg,algo=rigid:step=2,type=seg,algo=bsplinesyn,slicewise=1,iter=3 \
+                           -qc ~/qc_singleSubj
+
+:Input arguments:
+   - ``-i`` : Source image. Here, we select the T2* version of the PAM50 template.
+   - ``-iseg`` : Segmentation for the source image. Here, we use the PAM50 segmented white matter volume, rather than the spinal cord volume. This allows us to account for both the cord and the gray matter shape.
+   - ``-d`` : Destination image.
+   - ``-dseg`` : Segmentation for the destination image. Here, we use the white matter segmentation for the same reasons as ``-iseg``.
+   - ``-initwarp`` : Warping field used to initialize the source image. Here, we supply the ``warp_template2anat.nii.gz`` file from the previous T2 registration. (See: :ref:`template-registration`)
+      - TODO: I'm concerned that the usage of the ``-initwarp`` flag is not clear enough. For users who are following this tutorial standalone, they may not have the full context of the previous tutorial, so the usage of ``-initwarp`` may appear to "come out of nowhere". But, even if they have the context, reusing intermediate results raises some questions:
+          - Is ``-initwarp`` optional or mandatory?
+          - What should the command look like without ``-initwarp``?
+          - Will the registration results be significantly worse without ``-initwarp``?
+          - In what cases should ``-initwarp`` be used, and in what cases can they perform registration directly?
+   - ``-initwarpinv``: Warping field used to initialize the destination image. Here, we supply the inverse warping field, ``warp_anat2template.nii.gz`` from the previous T2 registration. (See: :ref:`template-registration`)
+      - TODO: Same concerns as above.
+   - ``-param`` :
+      - TODO: Why are we using ``rigid`` specifically? The MT tutorial uses ``centermass`` instead... do we need to explain the discrepancy? (Not explained in SCT course.)
```

Comment:
> I would say that `rigid` includes rotation while `centermass` doesn't

Ahhh, I see. In that case, I wonder why we don't use `centermassrot`? Since, that is what is used by default for step 1.

Or, maybe a  better question would be "why don't we use the default parameters?"

Reply:
> If `centermassrot` is specified, it would de facto be a slicewise transformation (making it less regularized than the slicewise=0 option and thus more prone to misalignment in some slices in cases where the segmentation is not great).

Ahhh, interesting. This is a subtle detail I hadn't realized before. I have a few follow-up questions, then:

* I see now that [the "customing registration" page](https://spinalcordtoolbox.com/en/latest/user_section/tutorials/template-registration/customizing-registration.html#the-param-argument) mentions that `slicereg` is a slice-by-slice transformation, and that `centermassrot` is similar to slicereg. So, I guess the detail about slicewise transformations could be inferred? But, perhaps we should explicitly state that `centermassrot` is also a slice-by-slice transformation, too
* Does that mean that `slicewise=0`/`slicewise=1` has no effect if you're combining it with `slicereg` or `centermassrot`?
* If `centermassrot` is more prone to misalignment with poor segmentation, then why have we made it the default value for `param` for step 1 of `sct_register_to_template`?

Reply:
> Question: If `centermassrot` is more prone to misalignment with poor segmentation, then why have we made it the default value for `param` for step 1 of `sct_register_to_template`? (Just trying to understand the motivation for why someone would want to use certain algorithms over others.)

Another great question. We made it the default because:
- the SC segmentation is usually good enough for estimating a robust center of mass, whereas the WM segmentation (used in `gm-informed-t2s-template-registration.rst`) has "holes" in it (due to the GM) and the variable shape of the GM (eg possibly asymmetry) makes it a less reliable object for estimating a centerline;
- the `rigid` algorithm might not offer enough degrees of freedom for registering a new subject to the template (eg: in case of problems with the straightening, or in case the flag `-reference subject` is used, in which case no/minimum rostrocaudal regularization should be used).

These are kind of "advanced" considerations and I'm afraid that >90% of our novice users will get lost in these explanations.

> Also, just looking at our documentation, I see now that [the "customizing registration" page](https://spinalcordtoolbox.com/en/latest/user_section/tutorials/template-registration/customizing-registration.html#the-param-argument) mentions that `slicereg` is a slice-by-slice transformation, and that `centermassrot` is similar to slicereg. So, I guess the detail about slicewise transformations could be inferred? But, perhaps we should explicitly state that `centermassrot` is also a slice-by-slice transformation, too.

sure, we can definitely mention that.

Reply:
Thank you for taking the time to write out a more detailed explanation! I'm learning a lot. :smile: 

> These are kind of "advanced" considerations and I'm afraid that >90% of our novice users will get lost in these explanations.

Good point. Perhaps for the sake of keeping the tutorial simple, we can just leave the various commands as-is for now. But, in the future, try to summarize all of the more nuanced decisions in the "Advanced Usage" section of the registration command page. 

See also, #3095:

> We have existing "Advanced Usage Tips" pages for some commands. These pages are currently in the "Tutorials" section, but we could move these tips and tricks to each script's dedicated page. That way, the usage information for script is centralized in 1 location.